homeassistant/components/light/elan.py

```diff
@@ -0,0 +1,183 @@
+"""
+Support for the elan.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/xxxx/
+"""
+import asyncio
+import aiohttp
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_HS_COLOR,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR, Light)
+from homeassistant.components.light import  PLATFORM_SCHEMA
+from homeassistant.util import color as color_util
+from homeassistant.helpers import config_validation as cv
+
+#REQUIREMENTS = ['pyelan']
+
+_LOGGER = logging.getLogger(__name__)
+
+#MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+#MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(milliseconds=100)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required('url'): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the elan Light platform."""
+    if discovery_info is None:
+        return
+
+    url = discovery_info['url']
+
+    session = aiohttp.ClientSession()
```

Comment:
Don't create new aiohttp sessions. Use the home assistant helper function to get the session.

Reply:
ok. For the moment I have changed code to use helper.

Is the separation of communication into separate library really needed? At this momen it would be only warapper around aiohttp getpost.